src/test/kontrol/TermAuctionListInvariants.t.sol

```diff
@@ -0,0 +1,502 @@
+
+import "src/test/kontrol/Constants.sol";
+import "src/test/kontrol/RepoToken.sol";
+import "src/test/kontrol/RepoTokenListInvariants.t.sol";
+import "src/test/kontrol/TermAuction.sol";
+import "src/test/kontrol/TermAuctionOfferLocker.sol";
+import "src/test/kontrol/TermDiscountRateAdapter.sol";
+
+contract TermAuctionListInvariantsTest is RepoTokenListInvariantsTest {
+    using TermAuctionList for TermAuctionListData;
+
+    TermAuctionListData _termAuctionList;
+    address _referenceAuction;
+
+    function setUp() public {
+        // Make storage of this contract completely symbolic
+        kevm.symbolicStorage(address(this));
+
+        // We will copy the code of this deployed auction contract into all
+        // auctions in the list
+        _referenceAuction = address(new TermAuction());
+
+        // Initialize TermAuctionList of arbitrary size
+        _initializeTermAuctionList();
+    }
+    
+    /**
+     * Return the auction for a given offer in the list.
+     */
+    function _getAuction(bytes32 offerId) internal returns(address) {
+        return address(_termAuctionList.offers[offerId].termAuction);
+    }
+
+    /**
+     * Deploy & initialize RepoToken and OfferLocker with the same RepoServicer
+     */
+    function _newRepoTokenAndOfferLocker() internal returns (
+        RepoToken repoToken,
+        TermAuctionOfferLocker offerLocker
+    ) {
+        repoToken = new RepoToken();
+        repoToken.initializeSymbolic();
+        (, , address termRepoServicer,) = repoToken.config();
+
+        offerLocker = new TermAuctionOfferLocker();
+        offerLocker.initializeSymbolic(termRepoServicer);
+    }
+
+    /**
+     * Initialize _termAuctionList to a TermAuctionList of arbitrary size,
+     * comprised of offers with distinct ids.
+     */
+    function _initializeTermAuctionList() internal {
+        bytes32 previous = TermAuctionList.NULL_NODE;
+        uint256 count = 0;
+
+        while (kevm.freshBool() != 0) {
+            (RepoToken repoToken, TermAuctionOfferLocker offerLocker) =
+                _newRepoTokenAndOfferLocker();
+
+            // Assign each offer an ID based on Strategy._generateOfferId()
+            bytes32 current = keccak256(
+                abi.encodePacked(count, address(this), address(offerLocker))
+            );
+            // Register offer in offer locker
+            offerLocker.initializeSymbolicLockedOfferFor(current);
+
+            if (previous == TermAuctionList.NULL_NODE) {
+                _termAuctionList.head = current;
+            } else {
+                _termAuctionList.nodes[previous].next = current;
+            }
+
+            // Create sequential addresses to ensure that list is sorted
+            address auction = address(uint160(1000 + count));
+            // Etch the code of the auction contract into this address
+            vm.etch(auction, _referenceAuction.code);
+            TermAuction(auction).initializeSymbolic();
```

Comment:
Unfortunately, we can't make the auction addresses completely symbolic here because we need to be able to call functions in the auction contracts. So instead, we initialize sequential addresses to guarantee the sortedness of the list and `etch` the code of the `TermAuction` contract into them.

Note, however, that this means that we are not covering the cases where we have multiple offers in the list for the same auction. It would be impractical to cover all possible combinations with symbolic execution. For example, for a list with length 3 we would have 5 separate cases: all 3 offers for different auctions, 1 and 2 share an auction, 1 and 3 share an auction, 2 and 3 share an auction, all share a single auction. For length 4 this shoots up to 15.

Some options to complement this could be:
- Fuzz tests that generate different test cases with repeated auctions.
- Select a specific subset of cases and write additional symbolic tests for them.